### DISCUSSION TITLE
pods_api_post_save_pod_* actions need $pieces returned

### DISCUSSION THREAD

**Discussion by mastef** [pods-framework/pods#6834] [Q&A]
it seems in add_action('pods_api_post_save_pod_item_my_pod_name', ... the $pieces array should be returned, as the 2nd action that fires on that event receives an empty $pieces value 

worth noting in http://pods.io/docs/code/action-reference/pods_api_post_save_pod_item/ && http://pods.io/docs/code/action-reference/pods_api_post_save_pod_item_podname/

**Comment by Shelob9** [comment]
As we discussed in irc you are right. Everything that goes through pods_do_hook is a filter.

This means that 1433ec8 and f172dec, as well as the docs on Pods.io and the code examples and the new tutorials in code-library need to be fixed. Sigh.

**Comment by Shelob9** [comment]
Separate (ish) issue, if we change the action to a filter in 3.0 than this will break everyone's post_save actions that already exist, which is any one who read the messed up docs.

@pglewis Suggests in 3.0 we have both post_save actions and filters, but mark the action as deprecated as of 3.0. That way we can be doing it right, have accurate docs and not break backwards compatibility.

@sc0ttkclark Do you agree with this?

**Comment by Shelob9** [comment]
@sc0ttkclark Points out to me that if we use do_action in 3.0, that keeps it as an action as it was meant to be, and will resolve the issue with multiple returns of $pieces not getting picked up.

**Comment by pglewis** [comment]
Clearing the milestone for 3.0 focus
